Tests/Unit/NetworkServer/CupsUpdateInfoResponseHeaderTests.cs

```diff
@@ -0,0 +1,140 @@
+            {
+                SignatureKeyCrc = 0,
+                UpdateDataLength = 0,
+            };
+
+            using var memoryPool = MemoryPool<byte>.Shared.Rent(2048);
+
+            // Act
+            var serializedResponse = updateResponseHeader.Serialize(memoryPool.Memory.Span);
+
+            // Assert
+            Assert.True(serializedResponse.ToArray().All(b => b == 0));
+        }
+
+        [Fact]
+        public void Serialize_WithFirmwareUpdates()
+        {
+            // setting up the twin in such a way that there are only firmware updates
+            var signatureBytes = Convert.FromBase64String("ABCD");
+            var keyCRC = 6789U;
+            var updateDataLength = 100U;
+            var updateResponseHeader = new CupsUpdateInfoResponseHeader
+            {
+                UpdateSignature = signatureBytes,
+                SignatureKeyCrc = keyCRC,
+                UpdateDataLength = updateDataLength,
+            };
+
+            using var memoryPool = MemoryPool<byte>.Shared.Rent(2048);
+
+            // Act
+            var serializedResponse = updateResponseHeader.Serialize(memoryPool.Memory.Span).GetReader();
+
+            // Assert
+            Assert.Equal(0, serializedResponse.Read()); //no uri
+            Assert.Equal(0, serializedResponse.Read()); //no uri
+            Assert.Equal(0, serializedResponse.ReadUInt16LittleEndian()); //no cred
+            Assert.Equal(0, serializedResponse.ReadUInt16LittleEndian()); //no cred
+            Assert.Equal((uint)signatureBytes.Length + 4, serializedResponse.ReadUInt32LittleEndian());
+            Assert.Equal(keyCRC, serializedResponse.ReadUInt32LittleEndian());
+            Assert.Equal(signatureBytes, serializedResponse.Read(signatureBytes.Length));
+            Assert.Equal(updateDataLength, serializedResponse.ReadUInt32LittleEndian());
+        }
+
+        [Fact]
+        public void Serialize_WithCupsUriUpdates()
+        {
+            // setting up the twin in such a way that there are only cups uri updates
+            var updateResponseHeader = new CupsUpdateInfoResponseHeader
+            {
+                CupsUrl = new Uri(updateUriString)
+            };
+
+            using var memoryPool = MemoryPool<byte>.Shared.Rent(2048);
+
+            // Act
+            var serializedResponse = updateResponseHeader.Serialize(memoryPool.Memory.Span).GetReader();
+
+            // Assert
+            Assert.Equal(updateUriString.Length, serializedResponse.Read());
+            Assert.Equal(updateUriString, serializedResponse.ReadUtf8String(updateUriString.Length));
+            Assert.Equal(0, serializedResponse.Read());
+            Assert.Equal(0, serializedResponse.ReadUInt16LittleEndian()); //no cred
+            Assert.Equal(0, serializedResponse.ReadUInt16LittleEndian()); //no cred
+            Assert.Equal(0U, serializedResponse.ReadUInt32LittleEndian());
+            Assert.Equal(0U, serializedResponse.ReadUInt32LittleEndian());
```

Comment:
Suggest to add comments next to each 0-asserted length about help understand which field is being talked about:

```suggestion
            Assert.Equal(0, serializedResponse.Read()); //no tcUri
            Assert.Equal(0, serializedResponse.ReadUInt16LittleEndian()); //no cupsCred
            Assert.Equal(0, serializedResponse.ReadUInt16LittleEndian()); //no tcCred
            Assert.Equal(0U, serializedResponse.ReadUInt32LittleEndian()); // no sig + keyCRC
            Assert.Equal(0U, serializedResponse.ReadUInt32LittleEndian()); // no updData
```